0001-short-term-release-process/0001-release-process-short-term.md

```diff
@@ -0,0 +1,325 @@
+
+In our setup, we would have two permanent branches, `main` and `stable`, where `main` is for
+day-to-day development and `stable` is essentially for tracking releases we consider deployable.
+Like Gitflow, we would also have some temporary branches; these will be discussed in more detail
+later.
+
+### Merging
+
+Historically, our team has preferred `git rebase` to `git merge`; however, Gitflow is better
+supported by using `git merge`. Rebasing is fine before submitting upstream, but Gitflow involves
+merging commits between different branches. Due to the fact that rebasing rewrites the commit
+history, merging between branches can lead to commits that have the same content but different
+hashes. This can make merging more confusing than it needs to be and also cause the commit history
+to be littered with duplicates. The primary option for completing a PR should be to merge it in
+rather than rebase.
+
+## Release Cycle Overview
+
+This section provides an overview of a release cycle. Certain aspects here merit more detail, but
+those will be provided in subsequent sections.
+
+The cycle has the following phases and steps:
+
+* Internal development and testing phase (two weeks):
+    - Begins immediately after production deploy of previous wave
+    - Deploy previous stable version to `STG-02` **
+    - Feature branches are worked on and merged back into `main`
+    - Developers can deploy their own isolated testnets if necessary
+    - We can do quick `alpha` releases if necessary
+* Release candidate (RC) phase (one week):
+    - Create a `release-YYYY.MM.rc.1` branch from `main`. No new features will be accepted on this
+      branch.
+    - Bump version numbers, with an `rc.1` suffix applied.
+    - Build RC and release to Github as a public `pre-release` but with no published crates.
+    - Deploy RC to `STG-01`.
+    - Production of changelog begins.
+    - Invite community members to test against this network.
+    - Shu's metrics solution enables comparisons to be made to the previous stable release in `STG-02`.
+    - Fixes can be applied to the release branch, resulting in an `rc.2`. This will be
+      released/deployed/tested. Repeat if necessary.
+* Release and deploy phase (one/two days):
+    - On the release branch, modify version numbers to remove the `rc` pre-release specifier.
+    - Changelog finalised.
+    - Merge the release branch into `stable`.
+    - Merge the release branch into `main`.
+    - When ready, perform a release and tag of `stable` by kicking off a workflow. *
+    - Delete the release branch.
+    - Deploy to `PROD-01`/`02` and drain *
+    - Announce to users
+
+We can accommodate hotfixes at any point during the cycle.
+
+#### Uncertainties
+
+* Not completely sure about the duration of each phase. Any opinions?
+* We could deploy automatically, but do we need to coordinate with an announcement?
```

Comment:
I'd say only final release woudl need an announcement. Alpha/beta/rc could should go out as/when ready to keep the flow simpler.

Ideally it would allow for auto-updating if you've opted in to that release channel eg to keep the need for announcements there to a minimum. (Anything we want could/should just be added to the github release post?)

Reply:
In this proposal, we don't really have the notion of a persistent channel. If community members were to participate in either alpha or RC testing, they would just use specific version numbers. Just want to make sure we are both on the same page with that one.

Reply:
Yeh, that makes sense.

The "channel" concept is more like "what kind of releases would you like to auto-update to". With no auto updating, there is no "channel", people just run what they want manually.

Reply:
OK cool. Again, this is something that will probably be different for post-release, but for short term, it sounds like we are in agreement that there are no channels.